backend/src/main/java/ufrn/imd/boraPagar/user/UserService.java

```diff
@@ -137,10 +140,10 @@ public UserModel addFriend(String credential, String googleId) {
     
     public UserModel removeFriend(String credential, String googleId) {
         UserModel user = getExistingOrNewUserFromCredential(credential);
-        UserModel friend = userRepository.findByGoogleId(googleId);
+        Optional<UserModel> friend = userRepository.findByGoogleId(googleId);
 
-        if(user != null && friend != null && user.getFriends().contains(friend)) {
-            user.getFriends().remove(friend);
+        if(user != null && friend.isPresent() && user.getFriends().contains(friend.get())) {
+            user.getFriends().remove(friend.get());
```

Comment:
A segunda verificação se torna inútil, não? Caso não exista um usuário com o id passado o `friend.get()` vai lançar a exceção e não irá executar o resto do código, certo?

Apesar disso, acho que esse é um caso em que vale a pena deixar o `isPresent()` por motivação semântica. Concordam?

Reply:
@isaacmsl